src/notation/internal/notationactioncontroller.cpp

```diff
@@ -737,6 +737,38 @@ void NotationActionController::padNote(const Pad& pad)
     if (currentNotationElements()->msScore()->inputState().usingNoteEntryMethod(engraving::NoteEntryMethod::RHYTHM)) {
         playSelectedElement();
     }
+
+    std::string spokenNote;
+    switch (pad) {
+    case Pad::NOTE1:
+        spokenNote = trc("notation", "Whole note");
+        break;
+    case Pad::NOTE2:
+        spokenNote = trc("notation", "Half note");
+        break;
+    case Pad::NOTE4:
+        spokenNote = trc("notation", "Quarter note");
+        break;
+    case Pad::NOTE8:
+        spokenNote = trc("notation", "Eighth note");
+        break;
+    case Pad::NOTE16:
+        spokenNote = trc("notation", "Sixteenth note");
+        break;
+    case Pad::NOTE32:
+        spokenNote = trc("notation", "Thirty-second note");
+        break;
+    case Pad::NOTE64:
+        spokenNote = trc("notation", "Sixty-fourth note");
+        break;
+    default:
+        break;
+    }
```

Comment:
It would be good to move this `switch` to a separate function that takes a `Pad` enum value as argument and returns the name as a string.

However, I'm not sure where to put the new function. It could go in the current file, or it could go in [typesconv.cpp](https://github.com/musescore/MuseScore/blob/master/src/engraving/types/typesconv.cpp). But if it went in typesconv.cpp, you'd probably have to move `Pad` from [score.h](https://github.com/musescore/MuseScore/blob/326b06ccaabc77faa79db50d890bc0de12e9a941/src/engraving/dom/score.h#L159-L179) into [types.h](https://github.com/musescore/MuseScore/blob/master/src/engraving/types/types.h).

So I'd welcome a second opinion here.

Reply:
I would look for a solution to avoid duplicating button label text - this text can be obtained from the action.

I suggest moving the setTriggeredCommand setup to a higher level where we have access to the ActionCode and obtain the text from this code through IUiActionsRegister.